### PR TITLE
Implementing remap operator.

### PR DESCRIPTION
**Pull request by luitjens**
This operator takes an operator/tensor
and an index array and logically gathers elements from that index array.
This allows the user to logically select/broadcast elements from a
dimension of the tensor.  This operator could then be stored in a memory
backed tensor to create a linear ordering of an operator to pass into a
library backed routine like matmul or fft.

### CODE REVIEW

**Review by cliffburdick — APPROVED** [review]
Can you also add the function to docs_input?

File: include/matx_set.h [hunk 1]
```diff
@@ -81,14 +79,15 @@ class set : public BaseOp<set<T, RANK, Desc, Op>> {
    * @param op
    *   Input operator
    */
-  inline set(tensor_impl_t<T, RANK, Desc> &out, const Op op) : out_(out), op_(op)
+  inline set(T &out, const Op op) : out_(out), op_(op)
   {
+    static_assert(is_matx_op_lhs<T>() == true, "Invalid operator on LHS of set/operator=");
```
Comment (cliffburdick) [inline include/matx_set.h:84]: should this be named lvalue instead of lhs?

File: include/matx_tensor_ops.h [hunk 2]
```diff
@@ -617,6 +617,64 @@ inline
...
+    constexpr __MATX_INLINE__ __MATX_HOST__ __MATX_DEVICE__ index_t Size(int dim) const
+    {
+      if(dim == dim_)
+        return idx_.Size(0);
+      else 
+        return op_.Size(dim);
+    }
+    
+    template<typename R> 
+    inline auto operator=(R rhs) {
```
Comment (cliffburdick) [inline include/matx_tensor_ops.h:672]: by reference?

File: include/matx_set.h (hunk 1 above)
Comment (luitjens) [reply include/matx_set.h]: changed

File: include/matx_tensor_ops.h (hunk 2 above)
Comment (luitjens) [reply include/matx_tensor_ops.h]: made const reference

**Pull request merged**